src/dlsnode/storage/BucketFile.d

```diff
@@ -440,6 +440,66 @@ public class BucketFile: OutputStream
 
     /**************************************************************************
 
+        Reads data from the file not blocking the fiber if there's not enough
+        data, giving that opportunity to the user. In that case, user should
+        suspend the fiber, wait for it to be resumed and call this method again.
+
+        Params:
+            waiting_context = ContextAwaitingJob used to wake
+                the fiber when the read is completed
+            buf = a buffer to write the data read from the file
+            wait_for_data = indicator if the fiber should suspend and wait until
+                            resumed via `waiting_context`
+
+        Returns:
+            number of bytes read from the file, or -1 if this should be suspended,
```

Comment:
Ha, so it could actually just return `-1` and `wait_for_data` could be implemented on the higher layer (in methods returning void). Because right now this both returns `-1` and set `wait_for_data`